src/main/scala/beam/analysis/RideHailFleetAnalysis.scala

```diff
@@ -0,0 +1,393 @@
+package beam.analysis
+
+import beam.agentsim.agents.vehicles.BeamVehicleType
+import beam.agentsim.events.{ParkingEvent, PathTraversalEvent, RefuelSessionEvent}
+import beam.analysis.plots.GraphAnalysis
+import beam.router.Modes.BeamMode
+import beam.sim.BeamServices
+import beam.sim.metrics.SimulationMetricCollector
+import beam.sim.metrics.SimulationMetricCollector.SimulationTime
+import org.matsim.api.core.v01.Id
+import org.matsim.api.core.v01.events.Event
+import org.matsim.core.controler.events.IterationEndsEvent
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+case class EventStatus(start: Double, end: Double, eventType: String, nextType: Option[String] = None)
+
+class RideHailFleetAnalysis(beamServices: BeamServices) extends GraphAnalysis {
+
+  val resolutionInSeconds = 60
+  val lastHour = 25
+  val timeBins = 0 until lastHour * 3600 by resolutionInSeconds
+  var processedHour = 0
+
+  val states = List(
+    "driving-full",
+    "driving-reposition",
+    "driving-topickup",
+    "driving-tocharger",
+    "queuing",
+    "charging",
+    "idle",
+    "offline",
+    "parked"
+  )
+
+  val keys = states.zipWithIndex.toMap
+
+  import RefuelSessionEvent._
+
+  val rideHailEvCav = mutable.Map[String, ArrayBuffer[Event]]()
+  val ridehailEvNonCav = mutable.Map[String, ArrayBuffer[Event]]()
+  val rideHailNonEvCav = mutable.Map[String, ArrayBuffer[Event]]()
+  val rideHailNonEvNonCav = mutable.Map[String, ArrayBuffer[Event]]()
+
+  override def processStats(event: Event): Unit = {
+    event match {
+      case refuelSessionEvent: RefuelSessionEvent =>
+        if (refuelSessionEvent.getAttributes.get(ATTRIBUTE_ENERGY_DELIVERED).toDouble > 0.0) {
+          val vehicle = refuelSessionEvent.getAttributes.get(RefuelSessionEvent.ATTRIBUTE_VEHICLE_ID)
```

Comment:
Replace `refuelSessionEvent.getAttributes` by directly accessing `refuelSessionEvent.energyInJoules` and `refuelSessionEvent.vehId` (seems like you need make one of the visible). The reason is because `refuelSessionEvent.getAttributes` will allocate HashMap and you also need to convert string to double

Reply:
done